### ISSUE TITLE
pulumi convert fails 

### ISSUE THREAD

**Issue by matt-slalom** [pulumi/pulumi#13173]
### What happened?

Pulumi convert command errored and requested I file a bug report

### Expected Behavior

I expected a successful conversion from TF to Pulumi

### Steps to reproduce

```shell
% pulumi convert --from terraform --language typescript --out pulumi
Converting from terraform...
warning: /Users/username/Work/path/config/vpc/vpc_subnet_demo/variables.tf:16,1-15: Failed to evaluate provider config; Could not evaluate expression for aws:region
warning: /Users/username/Work/path/config/vpc/vpc_subnet_demo/variables.tf:25,1-15: Provider alias not supported; Provider aliases are not supported, ignoring aws=us-west-2
warning: /Users/username/Work/path/config/vpc/vpc_subnet_demo/variables.tf:30,1-15: Provider alias not supported; Provider aliases are not supported, ignoring aws=us-west-1
warning: /Users/username/Work/path/config/vpc/vpc_subnet_demo/variables.tf:35,1-15: Provider alias not supported; Provider aliases are not supported, ignoring aws=us-east-1
Converting to nodejs...
================================================================================
The Pulumi CLI encountered a code generation error. This is a bug!
We would appreciate a report: https://github.com/pulumi/pulumi/issues/
Please provide all of the below text in your report.
================================================================================
Pulumi Version:   v3.71.0
error: variables.pp:42,5-6: Missing key/value separator; Expected an equals sign ("=") to mark the beginning of the attribute value.
error: could not generate output program
```

### Output of `pulumi about`

```shell
% pulumi about                                                                                                            ✘ 255 
CLI          
Version      3.71.0
Go Version   go1.20.4
Go Compiler  gc

Host     
OS       darwin
Version  13.4
Arch     arm64

Pulumi locates its logs in /var/folders/s_/t_kpxywj2wbfzc5x835tcrsc0000gn/T/ by default
warning: Failed to read project: no Pulumi.yaml project file found (searching upwards from /Users/username/Work/path). If you have not created a project yet, use `pulumi new` to do so: no project file found
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by matt-slalom** [comment]
```shell
% pulumi plugin ls
NAME        KIND       VERSION  SIZE    INSTALLED    LAST USED
aws         resource   5.41.0   542 MB  3 hours ago  3 hours ago
aws-native  resource   0.8.0    61 MB   3 hours ago  3 hours ago
terraform   converter  1.0.3    58 MB   3 hours ago  3 hours ago

TOTAL plugin cache size: 660 MB
```

**Comment by abhinav** [comment]
Are you able to share the Terraform program or a subset of it for us to debug this?

**Comment by abhinav** [comment]
Thanks, @matt-slalom. I was able to repro the error with the following config:

```hcl
locals {
  vpc_to_availability_zones_west = {
    0 = {
      public   = "1.2.3.4"
    }
  }
}
```

This produces an invalid PCL right now (the intermediate language from which we generate the language-specific code):

```
vpcToAvailabilityZonesWest = {
  0 0 = {
    public = "1.2.3.4"
  }
}
```

We'll work on fixing this. Thanks again, for making the repro easy!

**Comment by Zaid-Ajaj** [comment]
I've implemented a [fix](https://github.com/pulumi/pulumi-terraform-bridge/pull/1227) in the converter and will let you know as soon as we publish a new version of it 🚀

**Comment by Zaid-Ajaj** [comment]
This issue is fixed by https://github.com/pulumi/pulumi-terraform-bridge/pull/1227 and is available in pulumi-converter-terraform v1.0.4
